### PR TITLE
`azurerm_kusto_cluster` - support for `enable_disk_encryption` and `enable_streaming_ingest`

### LINKED ISSUE #5713: Support for enabling encryption in azurerm_kusto_cluster
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The azurerm_kusto_cluster resource needs to allow cluster encryption to be enabled.

Based on other settings in azurerm resources which follow the pattern of enable_xx_encryption = true|false, something like:
enable_cluster_encryption = true
or simply:
enable_encryption = true   (since it's already in the cluster resource definition)
would be fine.

Currently, you have to enable it manually after creating the cluster through terraform.
There may be other options like this that aren't currently exposed in this resource, but this is what stands out to me right now.

### New or Affected Resource(s)

* azurerm_kusto_cluster

### Potential Terraform Configuration

```hcl
resource "azurerm_kusto_cluster" "cluster" {
  name = var.name
  location  = var.location
  resource_group_name = var.resource_group_name

  enable_encryption = true

  sku {
    capacity = 2
    name     = "Standard_D11_v2"
  }
}
```

### PR DESCRIPTION
**Pull request by aqche**
Fixes #5713

Adds the `enable_disk_encryption` option to the `azurerm_kusto_cluster` resource.

```
--- PASS: TestAccAzureRMKustoCluster_basic (1533.92s)
--- PASS: TestAccAzureRMKustoCluster_withTags (1959.20s)
--- PASS: TestAccAzureRMKustoCluster_sku (2519.03s)
--- PASS: TestAccAzureRMKustoCluster_enableDiskEncryption (2896.44s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/kusto/tests 2896.565s 
```

### CODE REVIEW

**Comment by aqche** [comment]
updated to include the work by @njuCZ in PR #5858

```
--- PASS: TestAccAzureRMKustoCluster_basic (1897.19s)
--- PASS: TestAccAzureRMKustoCluster_withTags (1989.38s)
--- PASS: TestAccAzureRMKustoCluster_sku (2444.75s)
--- PASS: TestAccAzureRMKustoCluster_propsUpdate (2463.38s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/kusto/tests 2464.766s
```

File: azurerm/internal/services/kusto/resource_arm_kusto_cluster.go [hunk 1]
```diff
@@ -122,13 +132,18 @@ func resourceArmKustoClusterCreateUpdate(d *schema.ResourceData, meta interface{
...
+	enableStreamingIngest := d.Get("enable_streaming_ingest").(bool)
 
 	sku, err := expandKustoClusterSku(d)
 	if err != nil {
 		return err
 	}
 
-	clusterProperties := kusto.ClusterProperties{}
+	clusterProperties := kusto.ClusterProperties{
+		EnableDiskEncryption:  utils.Bool(enableDiskEncryption),
```
Comment (katbyte) [inline azurerm/internal/services/kusto/resource_arm_kusto_cluster.go]: We could do this here
```suggestion
		EnableDiskEncryption:  utils.Bool(d.Get("enable_disk_encryption").(bool)),
```

File: azurerm/internal/services/kusto/tests/resource_arm_kusto_cluster_test.go [hunk 2]
```diff
@@ -30,6 +30,35 @@ func TestAccAzureRMKustoCluster_basic(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMKustoCluster_propsUpdate(t *testing.T) {
```
Comment (katbyte) [inline azurerm/internal/services/kusto/tests/resource_arm_kusto_cluster_test.go]: Could we just use 
```suggestion
func TestAccAzureRMKustoCluster_update(t *testing.T) {
```

for consistency

File: azurerm/internal/services/kusto/tests/resource_arm_kusto_cluster_test.go [hunk 3]
```diff
@@ -176,6 +205,28 @@ resource "azurerm_kusto_cluster" "test" {
 `, data.RandomInteger, data.Locations.Primary, data.RandomString)
 }
 
+func testAccAzureRMKustoCluster_propsUpdate(data acceptance.TestData) string {
```
Comment (katbyte) [inline azurerm/internal/services/kusto/tests/resource_arm_kusto_cluster_test.go]: Could we make this a complete test, so we can then go basic -> complete -> basic in the update test
```suggestion
func testAccAzureRMKustoCluster_update(data acceptance.TestData) string {
```

File: azurerm/internal/services/kusto/tests/resource_arm_kusto_cluster_test.go [hunk 4]
```diff
@@ -30,6 +30,35 @@ func TestAccAzureRMKustoCluster_basic(t *testing.T) {
 	})
 }
```
Comment (katbyte) [inline azurerm/internal/services/kusto/tests/resource_arm_kusto_cluster_test.go]: Could we have a complete test here as described before

File: azurerm/internal/services/kusto/tests/resource_arm_kusto_cluster_test.go [hunk 5]
```diff
@@ -30,6 +30,35 @@ func TestAccAzureRMKustoCluster_basic(t *testing.T) {
...
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMKustoCluster_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMKustoClusterExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "enable_disk_encryption", "false"),
+					resource.TestCheckResourceAttr(data.ResourceName, "enable_streaming_ingest", "false"),
+				),
+			},
+			{
```
Comment (katbyte) [inline azurerm/internal/services/kusto/tests/resource_arm_kusto_cluster_test.go]: Could wedo an import check here

File: azurerm/internal/services/kusto/tests/resource_arm_kusto_cluster_test.go [hunk 6]
```diff
@@ -30,6 +30,35 @@ func TestAccAzureRMKustoCluster_basic(t *testing.T) {
...
+				Config: testAccAzureRMKustoCluster_propsUpdate(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMKustoClusterExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "enable_disk_encryption", "true"),
+					resource.TestCheckResourceAttr(data.ResourceName, "enable_streaming_ingest", "true"),
+				),
+			},
+			data.ImportStep(),
+		},
```
Comment (katbyte) [inline azurerm/internal/services/kusto/tests/resource_arm_kusto_cluster_test.go]: And then here go back to basic

**Comment by aqche** [comment]
@katbyte thanks for the review! made the changes you suggested.

```
=== RUN   TestAccAzureRMKustoCluster_update
=== PAUSE TestAccAzureRMKustoCluster_update
=== CONT  TestAccAzureRMKustoCluster_update
--- PASS: TestAccAzureRMKustoCluster_update (3277.49s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/kusto/tests 3277.525s
```

**Pull request merged**